apps/site/pages/docs/building-sections/overrides/advanced-use-cases.mdx

```diff
@@ -0,0 +1,78 @@
+# Advanced use case: Overriding sections with images
+
+This guide demonstrates an advanced use case for `getOverriddenSection` in FastStore, focusing on customizing the icon of the Alert section to display an image instead.
+
+## Context
+
+- You want to display an image alongside the text in your alert messages.
+- The native Alert section provides an `icon` prop for customization, but you want more control and flexibility over the displayed content.
+
+## Implementation
+
+By leveraging `getOverriddenSection`, you can create a custom component called `AlertWithImage` that achieves the desired functionality.
+
+1. Import necessary components:
+
+   ```tsx
+   import { getOverriddenSection } from "@faststore/core";
+   import Image from "next/image";
+   import { useMemo } from "react";
+   ```
+
+2. Define the `AlertWithImageProps` interface:
+
+   ```tsx
+   interface AlertWithImageProps extends Omit<React.ComponentProps<ReturnType<typeof getOverriddenSection<"Alert">>>, "icon"> {
```

Comment:
@icazevedo, the Alert component here keeps showing an error and the message "Type 'string' does not satisfy the constraint". I've changed it to `AlertSection`, but still not working. Not sure if this is the reason, but I can't reproduce the outcome. Can you help here?

Reply:
Put `typeof AlertSection` replacing `ReturnType<typeof getOverriddenSection<"Alert">>` and it should work!